chia/server/ws_connection.py

```diff
@@ -335,33 +335,37 @@ async def send_messages(self, messages: List[Message]):
         for message in messages:
             await self.outgoing_queue.put(message)
 
+    async def _wait_and_retry(self, msg: Message, queue: asyncio.Queue):
+        try:
+            await asyncio.sleep(1)
+            await queue.put(msg)
+        except Exception as e:
+            self.log.debug(f"Exception {e} while waiting to retry sending rate limited message")
+            return
+
     async def _send_message(self, message: Message):
         encoded: bytes = bytes(message)
         size = len(encoded)
         assert len(encoded) < (2 ** (LENGTH_BYTES * 8))
+        rate_limit_exception_reason = None
+        if is_localhost(self.peer_host):
+            rate_limit_exception_reason = "peer is localhost"
+        # TODO: fix this special case. This function has rate limits which are too low.
+        if ProtocolMessageTypes(message.type) == ProtocolMessageTypes.respond_peers:
+            rate_limit_exception_reason = "we are gossiping peers"
+
         if not self.outbound_rate_limiter.process_msg_and_check(message):
-            if not is_localhost(self.peer_host):
+            if rate_limit_exception_reason is None:
                 self.log.debug(
                     f"Rate limiting ourselves. message type: {ProtocolMessageTypes(message.type).name}, "
                     f"peer: {self.peer_host}"
                 )
-
-                # TODO: fix this special case. This function has rate limits which are too low.
-                if ProtocolMessageTypes(message.type) != ProtocolMessageTypes.respond_peers:
-
-                    async def wait_and_retry(msg: Message, queue: asyncio.Queue):
-                        try:
-                            await asyncio.sleep(1)
-                            await queue.put(msg)
-                        except Exception as e:
-                            self.log.debug(f"Exception {e} while waiting to retry sending rate limited message")
-                            return
-
-                asyncio.create_task(wait_and_retry(message, self.outgoing_queue))
```

Comment:
the original problem was that this line was supposed to be indented one more step, right?

Reply:
yes

Reply:
right